SS2-Project/Assets/Starstorm2/ContentClasses/SS2Equipment.cs

```diff
@@ -16,14 +16,52 @@ namespace SS2
     /// </summary>
     public abstract class SS2Equipment : IEquipmentContentPiece
     {
-        public abstract NullableRef<List<GameObject>> ItemDisplayPrefabs { get; }
+        public EquipmentAssetCollection AssetCollection { get; private set; }
+
+        public NullableRef<List<GameObject>> ItemDisplayPrefabs;
+        NullableRef<List<GameObject>> IEquipmentContentPiece.ItemDisplayPrefabs => ItemDisplayPrefabs;
         EquipmentDef IContentPiece<EquipmentDef>.Asset => EquipmentDef;
-        public abstract EquipmentDef EquipmentDef { get; }
 
-        public abstract bool Execute(EquipmentSlot slot);
+        public EquipmentDef EquipmentDef;
+
+        public abstract SS2AssetRequest<T> AssetRequest<T>() where T : UnityEngine.Object;
         public abstract void Initialize();
         public abstract bool IsAvailable(ContentPack contentPack);
-        public abstract IEnumerator LoadContentAsync();
+        public virtual IEnumerator LoadContentAsync()
+        {
+            SS2AssetRequest<UnityEngine.Object> request = AssetRequest<UnityEngine.Object>();
+
+            request.StartLoad();
+            while (!request.IsComplete)
+                yield return null;
+
+            if ((EquipmentAssetCollection)request.Asset)
+            {
+                AssetCollection = (EquipmentAssetCollection)request.Asset;
+
+                EquipmentDef = AssetCollection.equipmentDef;
+                ItemDisplayPrefabs = AssetCollection.itemDisplayPrefabs;
+
+                OnAssetCollectionLoaded(AssetCollection);
+            }
+            else if ((EquipmentDef)request.Asset)
```

Comment:
i'm around 90% sure this would throw an exception if the Asset is not of type EquipmentDef, I think it'd be better to use pattern matching here.

`if (reques.Asset is EquipmentDef eqpDef)`